distributed/shuffle/_worker_extension.py

```diff
@@ -315,6 +315,10 @@ def __init__(
             memory_limiter_comms=memory_limiter_comms,
             memory_limiter_disk=memory_limiter_disk,
         )
+        from dask.array.core import normalize_chunks
+
+        old = normalize_chunks(old)
```

Comment:
The identity check on `np.nan` doesn't survive serialization, so we have to normalize here again. @wence- had a very good when suggesting to switch the sentinel value (https://github.com/dask/dask/pull/10157#discussion_r1203663937).

Reply:
I suggest to add this as an in-code comment